src/events/reaction_add.rs

```diff
@@ -38,7 +38,21 @@ pub async fn reaction_add(ctx: client::Context, event: Reaction) -> Result<()> {
             }
         }
     }
-    handle_emoji_logging(ctx, event).await?;
+    if msg
+        .reactions
+        .iter()
+        .find_map(|x| {
+            if x.reaction_type == event.emoji && x.count == 1 {
+                Some(true)
+            } else {
+                None
+            }
+        })
+        .context("Could not find emoji for some reason")?
+        && !is_poll
```

Comment:
for a small performance improvement, the is_poll should probably be first here.
additionally, given how fucked discord is, the error message here might come up quite a lot - I think it might be sensible to just silence that here, and do `if let` instead of the context call here.